src/cascadia/ut_app/DynamicProfileTests.cpp

```diff
@@ -139,54 +140,54 @@ namespace TerminalAppUnitTests
         settings._profileGenerators.emplace_back(std::move(gen1));
 
         Profile p0, p1;
-        p0.SetName(L"profile0"); // this is _profiles.at(0)
-        p1.SetName(L"profile1"); // this is _profiles.at(1)
+        p0.Name(L"profile0"); // this is _profiles.at(0)
+        p1.Name(L"profile1"); // this is _profiles.at(1)
         settings._profiles.push_back(p0);
         settings._profiles.push_back(p1);
 
         settings._LoadDynamicProfiles();
         VERIFY_ARE_EQUAL(5u, settings._profiles.size());
 
-        VERIFY_ARE_EQUAL(L"profile0", settings._profiles.at(0).GetName());
-        VERIFY_IS_FALSE(settings._profiles.at(0)._guid.has_value());

```

Comment:
Maybe we should add some `HasGuid`, `HasSource`, `HasIcon` etc. helpers to Profile?

Reply:
I was considering that for a bit, actually. If we go that route, I think it would make sense to break up 
```c++
GETSET_PROPERTY(Windows::Foundation::IReference<guid>, Guid, nullptr);
```
into
```c++
// return/accept guid instead of IReference<guid>
bool HasGuid();
guid Guid();
void Guid(guid);
```
I ended up deciding against it because it'd be a bit of extra work. We could probably create a new `HASGETSET_OPTIONAL_PROPERTY` macro. On top of that, you can't have an `IReference<hstring>` so that's _extra_ annoying.

Reply:
> you can't have an `IReference<hstring>` so that's _extra_ annoying.

what the _H E C K_. Why do we live in this hellscape